Common/include/ad_structure.hpp

```diff
@@ -209,6 +240,17 @@ namespace AD{
    */
   void EndExtFunc();
 
+  /*!
+   * \brief Evaluates and saves gradient data from a variable.
+   * \param[in] data - where the information will be extracted from.
+   * \param[in] index - where obtained gradient data will be stored.
```

Comment:
It seems index is an output

Reply:
It's a reference to where we store the gradient information.. depends on the way you look at it I guess (?)

Reply:
the value of index gets set as a result of calling this function that's why it seems to be an output